File: BookStoreApi/Database/Models/Tag.cs
using Dapper.Contrib.Extensions;

namespace BookStoreApi.Database.Models
{
    [Table("Tags")]

    public class Tag
    {
        [Key]
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Url { get; set; } = string.Empty;
        public DateTime? CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
    }
}

File: BookStoreFront/src/app/models/tag.ts
import { Book, BookListResponse } from "./book";

export interface Tag {
  id: number;
  name: string;
  url: string;
  createdAt: Date;
  updatedAt: Date;
  books: Book[];
}

export interface TagListResponse {
  tags: Tag[];
  pagination: TagPaginationInfo;
}

export interface CreateTagRequest {
  name: string;
  url: string;
}

export interface UpdateTagRequest {
  name: string;
  url: string;
}

export interface TagDetails {
  tag: Tag;
  books: BookListResponse;
}

export interface TagPaginationInfo {
  totalCount: number;
  pageSize: number;
  pageNumber: number;
  totalPages: number;
}
